### DISCUSSION TITLE
Issue with Japanese language (nova-2)

### DISCUSSION THREAD

**Discussion by maxdubakov** [deepgram/community#796] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

The response returned by the nova-2 model for pre-recorded STT file in Japanese inserts spaces after words (when in Japanese there are generally no spaces between words, only after periods).

Second issue is that it sometimes would put period from the previous sentence in the next one.

Here's a sample of the response:

```
"paragraphs": [
                                {
                                    "sentences": [
                                        {
                                            "text": "皆 さ ん こんにちは 長 井 真 和 です。 イージー ジャパニーズ は 仙 台 から お 送り し ます 。 今日 は 街 の 人 た ち に 日 本 に つい て どう 思っ て いる か 、 行っ て みたい と 思い ます 。 今日 です ね 、日本 に つい て 、日本 の い い ところ に つい て",
                                            "start": 1.76,
                                            "end": 30.205
                                        }
                                    ],
                                    "speaker": 0,
                                    "num_words": 61,
                                    "start": 1.76,
                                    "end": 30.205
                                },
                                {
                                    "sentences": [
                                        {
                                            "text": "。はい。",
                                            "start": 30.205,
                                            "end": 30.705
                                        }
                                    ],
                                    "speaker": 1,
                                    "num_words": 1,
                                    "start": 30.205,
                                    "end": 30.705
                                }, [...]
``` 

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?paragraphs=true&punctuate=true&smart_format=true&model=nova-2&detect_language=true&utterances=false&diarize=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

e2e4f56d-569d-4458-be48-5aa9afab184a

### If possible, please attach an example audio file to reproduce the issue.

[y2mate.com - Easy Japanese 1  Typical Japanese.mp3.zip](https://github.com/user-attachments/files/15531686/y2mate.com.-.Easy.Japanese.1.Typical.Japanese.mp3.zip)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@maxdubakov  I'm curious to know if you try this request with specifying the `ja` language parameter instead of relying on language detection if you get a different experience?

You can also try removing paragraphs & punctuate as you've applied smart format and that functionality is included. 

```https://api.deepgram.com/v1/listen?&smart_format=true&model=nova-2&utterances=false&diarize=true&language=ja```

**Comment by maxdubakov** [comment]
Sure, I have tried this
`https://api.deepgram.com/v1/listen?smart_format=true&model=nova-2&utterances=false&diarize=true&language=ja`

And interestingly enough, in results.channels.alternatives.transcript I have recieved a proper formatting:
`"transcript": "皆さんこんにちは長井真和です。イージージャパニーズは仙台からお送りします。`

However, paragraphs.sentences still looks the same (with spaces) and if I include diarization, it will start some sentences with period as well.

`"paragraphs": {
    "paragraphs": [
        {
            "sentences": [
                {
                    "text": "皆 さ ん こんにちは...",
                    "start": 1.76,
                    "end": 30.205
                }
            ],
            "speaker": 0,
            "num_words": 61,
            "start": 1.76,
            "end": 30.205
        },
        {
            "sentences": [
                {
                    "text": "。はい。",
                    "start": 30.205,
                    "end": 30.705
                }
            ],
            "speaker": 1,
            "num_words": 1,
            "start": 30.205,
            "end": 30.705
        },`

**Comment by SandraRodgers** [comment, marked as answer]
The paragraphs feature only works on languages in which **words are delimited by spaces** (see https://developers.deepgram.com/docs/paragraphs). Since, as you said, in Japanese there are generally no spaces between words, only after periods, the paragraphs feature will not work as you want; it will create the spaces between the hiragana/katakana (etc) symbols.

**Reply by maxdubakov** [reply]
Oh, I see, thank You a lot!
